web-module/src/main/webapp/resources/structEditJS/dateTime/celements-date-time-field.js

```diff
@@ -162,317 +162,352 @@
+        && (!split[2] || (!isNaN(year) && (year > 100) && (year <= 9999)))) {
+        const curDate = new Date();
+        const date = new Date(year || curDate.getFullYear(), (month || (curDate.getMonth() + 1)) - 1,
+          day || curDate.getDate());
+        validated = $j.format.date(date, 'dd.MM.y');
+      }
+      console.debug("dateFieldValidator - to", validated);
+      return validated;
+    }
 
-      createTimePickerField: function(timeInputField, configObj = {}) {
-        const _me = this;
-        const pickerConfigObj = Object.assign({
-          'allowBlank': true,
-          'datepicker': false,
-          'format': 'H:i',
-          'step': 30
-        }, configObj);
-        return new CELEMENTS.structEdit.DateOrTimeFieldPicker(timeInputField, '.CelTimePicker',
-          "HH:mm", pickerConfigObj, _me._timeFieldValidator);
-      },
+    createTimePickerField(timeInputField, configObj = {}) {
+      const pickerConfigObj = Object.assign({
+        'allowBlank': true,
+        'datepicker': false,
+        'format': 'H:i'
+      }, configObj);
+      return new CELEMENTS.structEdit.DateOrTimeFieldPicker(timeInputField,
+        '.CelTimePicker', 'HH:mm', pickerConfigObj, this.#timeFieldValidator);
+    }
 
-      _timeFieldValidator: function(value) {
-        console.debug("timeFieldValidator - from", value);
-        value = (value || "").toString().trim().replace(/[\.,]/g, ':');
-        const split = value.split(":").filter(Boolean);
-        const hours = Number(split[0]);
-        let minutes = Number(split[1]);
-        if (minutes < 6 && split[1].trim().length == 1) {
-          minutes *= 10; // :5 -> 50 minutes
-        }
-        let validated = "";
-        if (value
-          && (!split[0] || (!isNaN(hours) && (hours >= 0) && (hours < 24)))
-          && (!split[1] || (!isNaN(minutes) && (minutes >= 0) && (minutes < 60)))) {
-          let date = new Date();
-          date.setHours(hours || 0);
-          date.setMinutes(minutes || 0);
-          validated = $j.format.date(date, "HH:mm");
-        }
-        console.debug("timeFieldValidator - to", validated);
-        return validated;
-      },
+    #timeFieldValidator(value) {
+      console.debug("timeFieldValidator - from", value);
+      value = (value || '').toString().trim().replace(/[\.,]/g, ':');
+      const split = value.split(':').filter(Boolean);
+      const hours = Number(split[0]);
+      let minutes = Number(split[1]);
+      if (minutes < 6 && split[1].trim().length == 1) {
+        minutes *= 10; // :5 -> 50 minutes
+      }
+      let validated = '';
+      if (value
+        && (!split[0] || (!isNaN(hours) && (hours >= 0) && (hours < 24)))
+        && (!split[1] || (!isNaN(minutes) && (minutes >= 0) && (minutes < 60)))) {
+        let date = new Date();
+        date.setHours(hours || 0);
+        date.setMinutes(minutes || 0);
+        validated = $j.format.date(date, 'HH:mm');

```

Comment:
Please replace `$j.format.date` with vanilla js replacement:
```suggestion
        validated = new Intl.DateTimeFormat('de-CH', {hour: '2-digit', minute:'2-digit' }).format((date);

```